<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TextEditor</class>
 <widget class="QMainWindow" name="TextEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TextEditor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>151</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>17</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionSelect_Font"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAboutQt"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionCut"/>
    <addaction name="separator"/>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionCopy"/>
   <addaction name="actionPaste"/>
   <addaction name="actionCut"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/icons/new.png</normaloff>:/icons/icons/new.png</iconset>
   </property>
   <property name="text">
    <string>Create</string>
   </property>
   <property name="toolTip">
    <string>Create new document</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="toolTip">
    <string>Close current window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="toolTip">
    <string>Close application</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionSelect_Font">
   <property name="text">
    <string>Select Font</string>
   </property>
   <property name="toolTip">
    <string>Select the display font</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionAboutQt">
   <property name="text">
    <string>AboutQt</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/icons/cut.png</normaloff>:/icons/icons/cut.png</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="toolTip">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/icons/copy.png</normaloff>:/icons/icons/copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="toolTip">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/icons/paste.png</normaloff>:/icons/icons/paste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="toolTip">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditUndo"/>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="toolTip">
    <string>Undo the last action</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditRedo"/>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="toolTip">
    <string>Redo the last action</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/icons/open.png</normaloff>:/icons/icons/open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open a document</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/icons/save.png</normaloff>:/icons/icons/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save document</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as...</string>
   </property>
   <property name="toolTip">
    <string>Save document as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>textEdit</sender>
   <signal>copyAvailable(bool)</signal>
   <receiver>actionCopy</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textEdit</sender>
   <signal>copyAvailable(bool)</signal>
   <receiver>actionCut</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textEdit</sender>
   <signal>undoAvailable(bool)</signal>
   <receiver>actionUndo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textEdit</sender>
   <signal>redoAvailable(bool)</signal>
   <receiver>actionRedo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCut</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>cut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>75</x>
     <y>104</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCopy</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>75</x>
     <y>104</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPaste</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>75</x>
     <y>104</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRedo</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>75</x>
     <y>104</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionUndo</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>75</x>
     <y>104</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
